addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -370,6 +370,57 @@ options.Class.include({
             },
         });
     },
+    /**
+     * @private
+     * @param {String} type either 'head' or 'footer'
+     */
+    async openCustomCodeDialog(previewMode, widgetValue, params) {
+        let websiteId;
+        this.trigger_up('context_get', {
+            callback: (ctx) => {
+                websiteId = ctx['website_id'];
+            },
+        });
+        const websites = await this._rpc({
+            model: 'website',
+            method: 'read',
+            args: [[websiteId], ['custom_code_head', 'custom_code_footer']],
+        });
+
+        const $content = $(core.qweb.render('website.custom_code_dialog_content', {
+            text: _.str.sprintf(_t('Enter code that will be added into the "%s" tag of every page of your site.'), widgetValue),
+        }));
+
+        const dialog = new Dialog(this, {
+            title: widgetValue === 'footer' ? 'Footer' : 'Head',
+            $content,
```

Comment:
To be honest, I don't like that syntax, I find it confusing especially when used with `key: value` around... but I am ok with it.